homeassistant/components/minecraft_server/__init__.py

```diff
@@ -58,11 +66,110 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
 
     # Clean up.
     server.stop_periodic_update()
-    hass.data[DOMAIN].pop(unique_id)
+    hass.data[DOMAIN].pop(config_entry_id)
 
     return unload_ok
 
 
+async def async_migrate_entry(hass: HomeAssistant, config_entry: ConfigEntry) -> bool:
+    """Migrate old config entry to a new format."""
+    _LOGGER.debug("Migrating from version %s", config_entry.version)
+
+    # 1 --> 2: Use config entry ID as base for unique IDs.
+    if config_entry.version == 1:
+        assert config_entry.unique_id
```

Comment:
I'd move this around so that we assign the local variable first and then assert that, so we don't need to access the same attribute more than once in the same scope.

Reply:
You mean
```
...
if config_entry.version == 1:
    old_unique_id = config_entry.unique_id
    assert old_unique_id
...
```
?
If so, I didn't understand what advantage this would have?

Reply:
Yes. The advantage is that we don't access the same attribute more than once. Attribute access is relatively more expensive accessing a local variable. It's just general code discipline.

Reply:
Ah I see, better performance. Got it, thanks! Will change it accordingly in another PR,